### ISSUE TITLE
Javascript error when loading a text file

### ISSUE THREAD

**Issue by rbatllet** [matecat/MateCat#2393]
I have installed the application locally but when loading a file I get a javascript error. The error message in the view is "Error: Invalid Upload Token. Check your browser, cookies must be enabled for this domain".

The javascript error is as follows:

upload.js:130852 Uncaught TypeError: Cannot read properties of undefined (reading 'split')

I am using the current version.

Cookies are fully enabled

Translated with www.DeepL.com/Translator (free version)
![Screenshot 2022-04-04 at 18 00 16](https://user-images.githubusercontent.com/3993987/161584692-9aacb773-aa12-4a0c-b1c3-107cce179073.png)

**Comment by riccio82** [comment]
Hi @rbatllet ,

Have you tried searching to see if anyone has had the same problem as you?

https://github.com/matecat/MateCat/issues/1575

**Comment by rbatllet** [comment]
All indications are that this problem has not yet been resolved.

**Comment by riccio82** [comment]
Are you using SSL?

**Comment by rbatllet** [comment]
Problem solved. The permissions for this file have to be as shown in the following image of the /home/matecat/matecat/inc directory:
![Screenshot 2022-04-06 at 16 18 18](https://user-images.githubusercontent.com/3993987/161996091-aa2485c0-ddca-43bc-b099-bc0a776724a1.png)

**Comment by rbatllet** [comment]
I have detected a problem. If I reload the page a couple of times, the file loading stops working. I have to reload again so that it can work again. 

![Screenshot 2022-04-07 at 11 41 04](https://user-images.githubusercontent.com/3993987/162170473-03c29253-5e06-472a-af7e-ccaaea231d66.png)

**Comment by jiakhan20** [comment]
Hi everyone,

I am facing the same issue as describe in this thread. Can anyone solve this issue?

**Comment by rbatllet** [comment]
@jiakhan20 Have you previously done this? -> https://github.com/matecat/MateCat/issues/2393#issuecomment-1090328354

**Comment by jiakhan20** [comment]
@rbatllet Yes, first I was facing permission problem. Now I am facing #2481 and #2393 issue. My logs regarding these errors are as follow:

/var/log/apache2/matecat/error.log
[Tue Jan 10 00:32:12.812193 2023] [:error] [pid 4646] [client 192.168.8.1:34302] PHP Notice:  Undefined index: upload_session in /home/matecat/matecat/lib/Utils/fileupload/upload.class.php on line 18, referer: https://projectmatecat.com/
[Tue Jan 10 00:32:12.813403 2023] [:error] [pid 4646] [client 192.168.8.1:34302] PHP Notice:  Undefined index: upload_session in /home/matecat/matecat/lib/Utils/fileupload/upload.class.php on line 19, referer: https://projectmatecat.com/
[Tue Jan 10 00:32:12.814255 2023] [:error] [pid 4646] [client 192.168.8.1:34302] PHP Notice:  Undefined index: upload_session in /home/matecat/matecat/lib/Utils/fileupload/upload.class.php on line 365, referer: https://projectmatecat.com/

/home/matecat/matecat/storage/log_archive/php_errors.txt
[11-Jan-2023 00:01:17 Europe/Rome] PHP Notice:  Undefined index: samesite in /home/matecat/matecat/lib/Utils/CookieManager.php on line 28
[11-Jan-2023 00:01:17 Europe/Rome] PHP Notice:  Undefined index: secure in /home/matecat/matecat/lib/Utils/CookieManager.php on line 30
[11-Jan-2023 00:01:17 Europe/Rome] PHP Notice:  Undefined index: httponly in /home/matecat/matecat/lib/Utils/CookieManager.php on line 31

There is no issue of permission and I tired to enable cookies for all website and also specifically  mention my domain as well.
